### DISCUSSION TITLE
How to handle DateTime fields of type Date on the client-side (convert ISO string)

### DISCUSSION THREAD

**Discussion by bennettdams** [prisma/prisma#6763] [Q&A]
How do you handle fields of type `Date` on your client-side?

Imagine a field with `DateTime` in the schema. The Prisma Client with TypeScript shows the field of type `Date` later, but when you fetch the data client-sided, it will be serialized to an (ISO) string. 

Here is a schema:

```
model Post {
  id             String       @id @default(cuid())
  createdAt      DateTime     @default(now())
  ..
}
```
The data is provided via a Next.js API route:

```ts
const post = await prisma.post.findUnique({ where: { id: postId } })
res.status(200).json(post)
```

...with a standard fetch:

```ts
async function fetchPost(postId: string): Promise<Post> {
  const response = await fetch(`${url}/${postId}`, {
    method: 'GET',
  })

  if (!response.ok) {
    throw new Error(response.statusText)
  }

  return await response.json()
}
```

...together with a custom hook and React Query:

```ts
export function usePost(postId: string | null) {
  const { data, isLoading, isError } = useQuery<Post>(
    [queryKeyPosts, postId],
    () => fetchPost(postId || ''),
    { enabled: !!postId }
  )
  const { create } = usePostsMutation()

  return { post: data || null, isLoading, isError, create: create.mutate }
}
```

I understand that I can easily convert the ISO string to a date, but where is the best place to do that?

This is a fairly simple example, but I'd need to do that for every schema model and every date field I'll ever use, right? There has to be a better way than manually creating dates after fetching on the client.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
